iOS_SDK/OneSignalSDK/Source/OneSignalHelper.m

```diff
@@ -504,7 +504,7 @@ + (void)handleNotificationAction:(OSNotificationActionType)actionType actionID:(
     
     [OneSignalTrackFirebaseAnalytics trackOpenEvent:result];
     
-    if (!notificationOpenedHandler)
+    if (!notificationOpenedHandler || displayType != OSNotificationDisplayTypeNotification)
```

Comment:
Instead of checking `displayType != OSNotificationDisplayTypeNotification`

Lets do `displayType == OSNotificationDisplayTypeSilent` so its a little more readable in that fact that `Silent` should not fire opened handler